### PR TITLE
[MUSA][16/N] Add MUSA backend support for layers

### PR DESCRIPTION
**Pull request by popsiclexu**
## Motivation
This PR is the 16th in a series of pull requests (tracked in https://github.com/sgl-project/sglang/issues/16565) to add full support for [Moore Threads](https://en.mthreads.com/) GPUs, leveraging MUSA (Meta-computing Unified System Architecture) to accelerate LLM inference.

dependencies： https://github.com/sgl-project/sglang/pull/17946

## Modifications
- Add forward_musa implementations for activation and quantization layers to support MUSA backend.
- Update condition checks to include MUSA alongside CUDA/HIP/XPU for kernel imports and execution paths.
- Modify deep_gemm integration to handle MUSA-specific imports and disable certain optimizations.
- Add MUSA-specific implementations for sampling, layernorm, MoE, and FP8 linear operations.

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/srt/layers/sampler.py [hunk 1]
```diff
@@ -103,8 +120,14 @@ def forward(
         logits = self._preprocess_logits(logits, sampling_info)
 
         if sampling_info.is_all_greedy:
-            # Use torch.argmax if all requests use greedy sampling
-            batch_next_token_ids = torch.argmax(logits, -1)
+            # XXX (MUSA): protect against empty logits tensor
```
Comment (froststeam) [inline python/sglang/srt/layers/sampler.py]: This can be removed to maintain consistency with the community.

File: python/sglang/srt/layers/quantization/fp8_utils.py [hunk 2]
```diff
@@ -52,6 +53,10 @@
 _is_cuda = is_cuda()
 _is_fp8_fnuz = is_fp8_fnuz()
 _is_gfx95_supported = is_gfx95_supported()
+_is_musa = is_musa()
+
+if _is_musa:
+    from sgl_kernel import musa_fused_gemv, musa_mudnn_w8a8_scaled_mm
```
Comment (froststeam) [inline python/sglang/srt/layers/quantization/fp8_utils.py]: Instead of _is_musa branch, using deep_gemm_wrapper.gemm_nt_f8f8bf16 could be a better option.

**Comment by popsiclexu** [comment]
move to https://github.com/sgl-project/sglang/pull/22774
